docs/ENCODING_BEST_PRACTICES.md

```
@@ -0,0 +1,301 @@
+
+### 3. Git Operations
+
+**DO:**
+```python
+# Git commit messages with emoji
+commit_msg = "Add feature ✅"
+repo.index.commit(commit_msg)  # Git handles UTF-8 by default
+
+# Reading git output
+log_output = repo.git.log('--oneline', encoding='utf-8')
+```
```

markdownlint-cli2 (0.18.1)

Fenced code blocks should be surrounded by blank lines

(MD031, blanks-around-fences)